src/python/pants/option/options_bootstrapper.py

```diff
@@ -3,255 +3,144 @@
-            config_files_products = [filecontent_for(p) for p in config_file_paths]
-            pre_bootstrap_config = Config.load(config_files_products, env=env)
-
-            initial_bootstrap_options = cls.parse_bootstrap_options(
-                env, bargs, pre_bootstrap_config
-            )
-            bootstrap_option_values = initial_bootstrap_options.for_global_scope()
-
-            # Now re-read the config, post-bootstrapping. Note the order: First whatever we bootstrapped
-            # from (typically pants.toml), then config override, then rcfiles.
-            full_config_sources = pre_bootstrap_config.sources()
-            if allow_pantsrc and bootstrap_option_values.pantsrc:
-                rcfiles = [
-                    os.path.expanduser(str(rcfile))
-                    for rcfile in bootstrap_option_values.pantsrc_files
-                ]
-                existing_rcfiles = [filecontent_for(p) for p in filter(os.path.exists, rcfiles)]
-                full_config_sources.extend(existing_rcfiles)
-
-            post_bootstrap_config = Config.load(
-                full_config_sources,
-                seed_values=bootstrap_option_values.as_dict(),
-                env=env,
-            )
-
-            bargs = cls._get_bootstrap_args(args)
-
-            # We need to set this env var to allow various static help strings to reference the
-            # right name (via `pants.util.docutil`), and we need to do it as early as possible to
-            # avoid needing to lazily import code to avoid chicken-and-egg-problems. This is the
-            # earliest place it makes sense to do so and is generically used by both the local and
-            # remote pants runners.
-            os.environ["__PANTS_BIN_NAME"] = munge_bin_name(
-                bootstrap_option_values.pants_bin_name, get_buildroot()
-            )
-
-            # TODO: We really only need the env vars starting with PANTS_, plus any env
-            #  vars used in env.FOO-style interpolation in config files.
-            #  Filtering to just those would allow OptionsBootstrapper to have a less
-            #  unwieldy __str__.
-            #  We used to filter all but PANTS_* (https://github.com/pantsbuild/pants/pull/7312),
-            #  but we can't now because of env interpolation in the native config file parser.
-            #  We can revisit this once the legacy python parser is no more, and we refactor
-            #  the OptionsBootstrapper and/or convert it to Rust.
-            env_tuples = tuple(sorted(env.items()))
-            return cls(
-                env_tuples=env_tuples,
-                bootstrap_args=bargs,
-                args=args,
-                config=post_bootstrap_config,
-            )
+        bootstrap_options = cls._create_bootstrap_options(args, env, allow_pantsrc)
+
+        # We need to set this env var to allow various static help strings to reference the
```

Comment:
I'm not sure we still need this feature. In the age of scie-pants I expect the script name to ~always be `pants`. But one thing at a time.